src/main/scala/is/hail/rvd/RVD.scala

```diff
@@ -137,7 +179,72 @@ trait RVD {
 
   def crdd: ContextRDD[RVDContext, RegionValue]
 
-  def rdd: RDD[RegionValue]
+  private[rvd] def stabilize(
+    unstable: ContextRDD[RVDContext, RegionValue],
+    codec: CodecSpec = RVD.memoryCodec
+  ): ContextRDD[RVDContext, Array[Byte]] = {
+    val enc = codec.buildEncoder(rowType) _
+    unstable.cmapPartitions { (ctx, it) =>
+      it.map(RVD.regionValueToBytes(enc, ctx))
+    }
+  }
+
+  private[rvd] def destabilize(
+    stable: ContextRDD[RVDContext, Array[Byte]],
+    codec: CodecSpec = RVD.memoryCodec
+  ): ContextRDD[RVDContext, RegionValue] = {
+    val dec = codec.buildDecoder(rowType)
+    stable.cmapPartitions { (ctx, it) =>
+      val rv = RegionValue(ctx.region)
+      it.map(RVD.bytesToRegionValue(dec, ctx.region, rv))
+    }
+  }
+
+  private[rvd] def stably(
+    f: ContextRDD[RVDContext, Array[Byte]] => ContextRDD[RVDContext, Array[Byte]]
+  ): ContextRDD[RVDContext, RegionValue] = destabilize(f(stabilize(crdd)))
+
+  private[rvd] def stably(
```

Comment:
Should this be on the object?

Reply:
I'd have to ship around the `rowType` which is used to construct the en/de-coders, so I'm inclined to say no